.github/workflows/test.yml

```diff
@@ -72,3 +72,44 @@ jobs:
           cd py4j-python
           # Java TLS tests are disabled until they can be fixed (refs #441)
           nosetests -v --exclude='.*java_tls_test.*'
+
+  test-doc:
+    name: Documentation build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout guidelines repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Install Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+
+      - name: Install Python packages
+        run: |
+          python3 -m pip install -r py4j-web/requirements-doc.txt
+          python3 -m pip list
+
+      - name: Test Sphinx build
+        # Build documentation as a test only in PRs.
+        if: github.event_name == 'pull_request'
+        run: |
+          cd py4j-web
+          make clean html
+
+      - name: Build and update documentation to GitHub pages
+        # Publishing documentation only for commits in the master branch.
+        if: github.event_name != 'pull_request'
+        run: |
+          cd py4j-web && make clean html && cd ..
+          mv py4j-web/_build/html ../  # Move out of the current repo.
+          git checkout gh-pages
+          rm -fr *  # Remove old site.
+          cp -r ../html/* .
+          git add .
+          # Maybe we should switch to Py4J dedicated bot in the future,
+          # instead of reusing Apache Spark's bot account.
```

Comment:
There is a github action to commit using the name and email of the author of the original commit (https://github.com/marketplace/actions/add-commit), but I guess by using a bot account you clearly show that this was an automated commit?

We should indeed create our own bot account (in the future, now this is fine)!